src/wui/interactive_player.cc

```diff
@@ -571,14 +571,22 @@ void InteractivePlayer::draw_map_view(MapView* given_map_view, RenderTarget* dst
 
 			const Image* player_image = nullptr;
 			float icon_scale = 0.7f;
+			float icon_opacity = 1.0f;
 
 			// Not all map starting positions pass the suitability test.
 			// TODO(tothxa): Make the editor at least use the same test. But manual changes would still
 			//               be possible.
 			for (unsigned p = map.get_nrplayers(); p != 0u; --p) {
 				if (map.get_starting_pos(p) == f->fcoords) {
-					player_image = playercolor_image(p - 1, icon_filename);
+					Widelands::Player* plr = gbase.get_player(p);
+					if (plr == nullptr || !plr->is_picking_custom_starting_position()) {
+						// Should have a HQ if finished picking, no need for the overlay
+						continue;
+					}
+					player_image = playercolor_image(plr->get_playercolor(), icon_filename);
```

Comment:
Do I understand the commit message correctly, `layercolor_image(p - 1, icon_filename);` is the players default colour and this line gives the current player colour? Then I am definitely for this.

Reply:
I tested it in widelands. It did show the default colour. So :+1: